GeosUtil/unitconv_mod.F90

```diff
@@ -148,39 +151,41 @@ MODULE UnitConv_Mod
 !\\
 ! !INTERFACE:
 !
-  SUBROUTINE Convert_Spc_Units( Input_Opt, State_Chm, State_Grid, State_Met, &
-                                outUnit,   RC,        origUnit )
+  SUBROUTINE Convert_Spc_Units( Input_Opt, State_Chm,     State_Grid,        &
+                                State_Met, mapping,       new_units,         &
+                                RC,        previous_units                   )
 !
 ! !USES:
 !
     USE TIMERS_MOD
 !
 ! !INPUT PARAMETERS:
 !
-    TYPE(OptInput),   INTENT(IN)    :: Input_Opt   ! Input Options object
-    TYPE(GrdState),   INTENT(IN)    :: State_Grid  ! Grid state object
-    TYPE(MetState),   INTENT(IN)    :: State_Met   ! Meteorology state object
-    INTEGER,          INTENT(IN)    :: outUnit     ! Desired output unit
+    TYPE(OptInput),   INTENT(IN)    :: Input_Opt      ! Input Options object
+    TYPE(GrdState),   INTENT(IN)    :: State_Grid     ! Grid state object
+    TYPE(MetState),   INTENT(IN)    :: State_Met      ! Meteorology state object
+    INTEGER,          INTENT(IN)    :: mapping(:)     ! Species map to modelId
+    INTEGER,          INTENT(IN)    :: new_units      ! Units to convert to
 !
 ! !INPUT/OUTPUT PARAMETERS:
 !
-    TYPE(ChmState),   INTENT(INOUT) :: State_Chm   ! Chemistry state object
+    TYPE(ChmState),   INTENT(INOUT) :: State_Chm      ! Chemistry state object
 !
 ! !OUTPUT PARAMETERS:
 !
-    INTEGER,          INTENT(OUT)   :: RC          ! Success or failure?
-    INTEGER,          OPTIONAL      :: origUnit    ! Units of input data
+    INTEGER,          INTENT(OUT)   :: RC             ! Success or failure?
+    INTEGER,          OPTIONAL      :: previous_units ! Previous units
 !
 ! !REMARKS:
 !  The purpose of optional output argument origUnit is to enable conversion
 !  back to the original units in a second call to Convert_Spc_Units.
 !  For example:
 !
-!      CALL Convert_Spc_Units( Input_Opt, State_Chm, State_Grid, State_Met, &
-!                              'kg/kg dry', RC, origUnit=origUnit )
+!      CALL Convert_Spc_Units( Input_Opt, State_Chm, State_Grid,       &
```

Comment:
This example doesn't look right. I think the first one should pass new_units and output previous_units, and the second one should set new_units to previous_units.

Reply:
@lizziel: This has now been updated in commit a8e07311.